src/Vts/IO/CustomBinaryStreamWriterOfT.cs

```diff
@@ -114,91 +127,40 @@ public void Write(IEnumerable<T> items)
         /// <summary>
         /// Closes the filestream and writes the accompanying .xml
         /// </summary>
-        public virtual void Close()
+        public void Close()
         {
             Dispose();
         }
 
-        #region IDisposable Members
-
-        private bool _disposed = false;
-        // Do not make this method virtual.
-        // A derived class should not be able to override this method.
-
         /// <summary>
-        /// Closes the file and writes the accompanying .xml
+        /// Internal method for disposing the IDisposable object
         /// </summary>
-        public void Dispose()
-        {
-            Dispose(true);
-            // This object will be cleaned up by the Dispose method.
-            // Therefore, you should call GC.SupressFinalize to
-            // take this object off the finalization queue
-            // and prevent finalization code for this object
-            // from executing a second time.
-            GC.SuppressFinalize(this);
-        }
-
-        // Dispose(bool disposing) executes in two distinct scenarios.
-        // If disposing equals true, the method has been called directly
-        // or indirectly by a user's code. Managed and unmanaged resources
-        // can be disposed.
-        // If disposing equals false, the method has been called by the
-        // runtime from inside the finalizer and you should not reference
-        // other objects. Only unmanaged resources can be disposed.
-        private void Dispose(bool disposing)
+        /// <param name="disposing">Specifies whether the object is currently being disposed</param>
+        protected virtual void Dispose(bool disposing)
         {
-            // Check to see if Dispose has already been called.
-            if (!this._disposed)
+            if (!disposedValue)

```

Comment:
Invert "if" to reduce nesting - if you merge before seeing this, we can fix later.

Reply:
Fixed :)